src/main/java/core/basesyntax/operations/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.Fruit;
+import core.basesyntax.Storage;
+import core.basesyntax.Transaction;
+
+public class SupplyOperation extends AbstactOperation implements StoreOperation {
+    public SupplyOperation(Storage storage) {
+        super(storage);
+    }
+
+    @Override
+    public void performOperation(Transaction transaction) {
```

Comment:
I don't understand why should I do it. 
In my application transaction is a kind of waybill, which "knows" what operation should be done with fruits and their quantity. And fruit only "knows" its type and expiration date. So if  StoreOperation.performOperation() will take Fruit instead of transaction it will be identical to Storage.addFruit(). And it has no sense. 
Or I should refactor class Fruit but I think fruits can't know what should be done with them. So I think that your suggestion doesn't fit into my logic

Reply:
Agree! Thanks for your explanation!